### PR TITLE
Adding grid support

### PR DESCRIPTION
**Pull request by danejo3**
The purpose of this PR is to integrate DRMAA grid support into YEAT.

Previously, YEAT would execute the snakemake workflow sequentially. To speed things up, parallel processing has been enabled. Users can adjust the grid execution parameters by passing them in YEAT's CLI. 

NOTE: Grid support has only been tested on SGE.

### CODE REVIEW

File: yeat/workflow/__init__.py [hunk 1]
```diff
@@ -17,14 +17,28 @@ def run_workflow(args):
...
+            snakefile,
+            config=config,
+            cores=args.threads,
+            dryrun=args.dry_run,
+            printshellcmds=True,
+            workdir=args.outdir,
+            local_cores=args.threads,
+            nodes=args.grid_limit,
+            drmaa=setup_grid_args(args),
+            drmaa_log_dir=str((Path(args.outdir) / "gridlogs").resolve()),
```
Comment (danejo3) [inline yeat/workflow/__init__.py:31]: In order to save the DRMAA snakemake log files, this must be a full path and not relative.

File: Makefile [hunk 2]
```diff
@@ -35,9 +35,13 @@ testbandage:
 testlinux:
 	pytest --cov=yeat -m linux
 
+## testgrid:    run only grid required automated tests
+testgrid:
+	pytest --cov=yeat -m grid --basetemp=tmp
+
```
Comment (danejo3) [inline Makefile:41]: To run grid only tests,

> make testgrid

Grid tests are not included in `make testall`.

**Review by danejo3 — COMMENTED** [review]
@standage 

This PR is ready for review! Thanks!

File: yeat/workflow/__init__.py (hunk 1 above)
Comment (standage) [reply yeat/workflow/__init__.py:31]: Good to know!

**Review by standage — CHANGES_REQUESTED** [review]
From a code review standpoint, this looks good. No concerns. I have not had a chance to test on the grid, but given the similarity to other workflows where we've implemented this approach, if you've had success with grid tests, I'm happy to accept that.

My only concern is your repeated claim that this is strictly SGE support. The whole point of DRMAA is that it's agnostic to the specific job scheduler used: SGE, PBS, Slurm, etc. It may be appropriate to say it's only been *tested* with SGE, but with DRMAA it should be straightforward to run on grids using a different scheduler. Please update the thread description, README, argument parser, etc., accordingly.

**Comment by danejo3** [comment]
@standage Updated the wording in various requested documents. Let me know if I missed anything. Thanks!

**Review by standage — APPROVED** [review]

**Pull request merged**